### DISCUSSION TITLE
guide for support Metamask with wallet connect v2

### DISCUSSION THREAD

**Discussion by andriirak** [WalletConnect/walletconnect-monorepo#2738] [Developer Support]
wallet connect v2 supports **multichain** wallets, but Matamask support **singlechain**, not clear how to work with this

Please provide a guide for support Metamask with wallet connect v2

**Comment by finessevanes** [comment]
Hey @andriirak not totally clear on what you're trying to do. Wha are you trying to build?

**Reply by andriirak** [reply]
hi, I'm dapp developer and I need to connect Metamask mobile wallet with walletconnect v2. In general it works, but have some errors (https://github.com/MetaMask/metamask-mobile/issues/6655, https://github.com/MetaMask/metamask-mobile/issues/6688) and misunderstading 
For example **EthereumProvider.init** has `chains` property that accept multiple chains, but Metamask supports only one chain, **how should I work with singlechain wallets?**

**Reply by finessevanes** [reply]
>but Metamask supports only one chain, how should I work with singlechain wallets?

You can just pass eth mainnet

**Reply by andriirak** [reply]
so if I'm using many blockchains in my app, but MM only one, I must always use  ` EthereumProvider.init({ chains: [1] })` .,right?
